coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala

```diff
@@ -0,0 +1,219 @@
+package com.socrata.datacoordinator.resources
+
+import java.io.{BufferedWriter, BufferedOutputStream}
+import java.nio.charset.StandardCharsets._
+import java.security.MessageDigest
+import javax.servlet.http.HttpServletResponse
+
+import com.rojoma.json.v3.ast.{JArray, JString, JNumber, JValue}
+import com.rojoma.json.v3.codec.JsonEncode
+import com.rojoma.json.v3.io.{CompactJsonWriter, JsonBadParse}
+import com.rojoma.json.v3.util.JsonArrayIterator
+import com.rojoma.simplearm.util._
+import com.socrata.datacoordinator.id.{UserColumnId, DatasetId}
+import com.socrata.datacoordinator.service.ServiceUtil._
+import com.socrata.datacoordinator.service._
+import com.socrata.datacoordinator.truth._
+import com.socrata.datacoordinator.truth.loader.DatasetDropper
+import com.socrata.datacoordinator.truth.metadata.{SchemaField, Schema}
+import com.socrata.datacoordinator.util.IndexedTempFile
+import com.socrata.datacoordinator.util.collection.UserColumnIdSet
+import com.socrata.http.server._
+import com.socrata.http.server.responses._
+import com.socrata.http.server.util.{EntityTag, Precondition}
+import org.joda.time.DateTime
+import org.joda.time.format.ISODateTimeFormat
+import com.socrata.http.server.implicits._
+import com.socrata.datacoordinator.common.util.DatasetIdNormalizer._
+import com.socrata.datacoordinator.external._
+
+
+
+case class DatasetResource(datasetId: DatasetId,
+                           tempFileProvider: () => IndexedTempFile,
+                           commandReadLimit: Long,
+                           processMutation: (DatasetId, Iterator[JValue], IndexedTempFile) => ProcessMutationReturns,
+                           deleteDataset: DatasetId => DatasetDropper.Result,
+                           datasetContents: (DatasetId, Option[String], CopySelector, Option[UserColumnIdSet],
+                             Option[Long], Option[Long], Precondition, Option[DateTime], Boolean) =>
+                             DatasetResource.datasetContentsFunc => Exporter.Result[Unit],
+                           withMutationScriptResults: (HttpResponse) => HttpResponse,
+                           formatDatasetId: DatasetId => String) extends ErrorHandlingSodaResource(formatDatasetId) {
+  override val log = org.slf4j.LoggerFactory.getLogger(classOf[DatasetResource])
```

Comment:
Just be aware that Scala Logging makes it very easy to produce a logger with the wrong name.  It won't happen in this case (because nothing will be extending `DatasetResource`) but `Logging` uses the class of the runtime instance to name the logger, which isn't necessarily the same as the class where the logging occurs, which is what you usually (almost always, really) want.